examples/gallery/src/icon-layer.html

```diff
@@ -0,0 +1,196 @@
+      const randomIndex = Math.floor(Math.random() * linkAngles.length);
+      const angle = linkAngles[randomIndex] * Math.PI / 180;
+      const x = Math.cos(angle) * linkDistance;
+      const y = Math.sin(angle) * linkDistance;
+      contributor['targetPosition'] = [ x, y ];
+      linkAngles.splice(randomIndex, 1);
+    }
+
+    function calculateMedian(inputArray) {
+      const array = [...inputArray];
+      if(array.length === 0) {
+        return 0;
+      }
+      array.sort((a, b) => a - b);
+      const medianIndex = Math.floor(array.length / 2);
+      if (array.length % 2) {
+        return array[medianIndex];
+      }
+      return (array[medianIndex - 1] + array[medianIndex]) / 2.0;
+    }
+
+    function calculateQuartiles(array) {
+      const secondQuartile = calculateMedian(array);
+      const firstHalfOfArray = array.filter(number => number < secondQuartile);
+      const secondHalfOfArray = array.filter(number => number > secondQuartile);
+
+      const firstQuartile = calculateMedian(firstHalfOfArray);
+      const thirdQuartile = calculateMedian(secondHalfOfArray);
+
+      return {firstQuartile, secondQuartile, thirdQuartile};
+    }
+
+    function updateTooltip({x, y, object}) {
+      const tooltip = document.getElementById('tooltip');
+      if (object) {
+        tooltip.style.top = `${y}px`;
+        tooltip.style.left = `${x}px`;
+        tooltip.innerHTML = `
+        <div><b>${object.login}</b></div>
+        <div>Contributions: ${object.contributions}</div>`;
+      } else {
+        tooltip.innerHTML = '';
+      }
+    }
+
+    function getSizeOfIcon(contributions, quartiles, ninetiethPercentile) {
+      const {firstQuartile, secondQuartile, thirdQuartile} = quartiles;
+
+        if (contributions <= firstQuartile) {
+          return 15;
+        } else if (contributions > firstQuartile && contributions <= secondQuartile) {
+          return 20;
+        } else if (contributions > secondQuartile && contributions <= thirdQuartile) {
+          return 25;
+        } else if (contributions > thirdQuartile && contributions <= ninetiethPercentile) {
+          return 32;
+        }
+        return 40;
+    }
+    
+
+    function renderLayers(data) {
+      const sourcePosition = [0, 0];
+      const listOfContributions = data.map(contributor => contributor.contributions);
+      const quartiles = calculateQuartiles(listOfContributions);
+      const ninetiethPercentile = [...listOfContributions]
+        .sort((a, b) => a - b)[Math.floor(0.90 * listOfContributions.length)];
+
+      const maxContributions = Math.max(...listOfContributions);
+      const minContributions = Math.min(...listOfContributions);
+
+      listOfContributions.forEach((number, index) => {
+        const linkDistance = 20 * ((number - minContributions) 
+        / (maxContributions - minContributions)) + 30;
+
+        addTargetPositionToContributor(data[index], linkDistance);
+      })
```

Comment:
Could extract the logic to get contributor nodes' positions to a separate function.